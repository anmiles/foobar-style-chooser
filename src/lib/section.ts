import ko from 'knockout';
import { Style } from './style';

export { Section };

class Section {
	title: string;
	styles: Style[];
	toString: ko.Computed<string>;

	constructor(title: string, styles: string[]) {
		this.title  = title;
		this.styles = styles.map((style) => new Style(style));
	}
}
