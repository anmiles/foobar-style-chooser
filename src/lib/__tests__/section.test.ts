import { Section } from '../section';

describe('src/lib/section', () => {
	it('should construct section', () => {
		const section = new Section('section1', [ 'style1', 'style2' ]);

		expect(section).toMatchSnapshot();
	});
});
