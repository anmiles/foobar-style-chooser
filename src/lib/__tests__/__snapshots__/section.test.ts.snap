// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/lib/section should construct section 1`] = `
Section {
  "styles": [
    Style {
      "enabled": [Function],
      "title": "style1",
    },
    Style {
      "enabled": [Function],
      "title": "style2",
    },
  ],
  "title": "section1",
}
`;
